enabler/src/de/schildbach/pte/ShProvider.java

```diff
@@ -164,4 +168,50 @@ protected NearbyLocationsResult nearbyStationsById(final String id, final int ma
+
+	protected static final Map<String, Style> STYLES = new HashMap<String, Style>();
+
+	static {
+		// Busse Kiel
+		putKielBusStyle("1", new Style(Style.parseColor("#7288af"), Style.WHITE));
+		putKielBusStyle("2", new Style(Style.parseColor("#50bbb4"), Style.WHITE));
+		putKielBusStyle("5", new Style(Style.parseColor("#f39222"), Style.WHITE));
+		putKielBusStyle("6", new Style(Style.parseColor("#aec436"), Style.WHITE));
+		putKielBusStyle("8", new Style(Style.parseColor("#bcb261"), Style.WHITE));
+		putKielBusStyle("9", new Style(Style.parseColor("#c99c7d"), Style.WHITE));
+		putKielBusStyle("11", new Style(Style.parseColor("#f9b000"), Style.WHITE));
+		putKielBusStyle("22", new Style(Style.parseColor("#8ea48a"), Style.WHITE));
+		putKielBusStyle("31", new Style(Style.parseColor("#009ee3"), Style.WHITE));
+		putKielBusStyle("32", new Style(Style.parseColor("#009ee3"), Style.WHITE));
+		putKielBusStyle("33", new Style(Style.parseColor("#009ee3"), Style.WHITE));
+		putKielBusStyle("34", new Style(Style.parseColor("#009ee3"), Style.WHITE));
+		putKielBusStyle("41", new Style(Style.parseColor("#8ba5d6"), Style.WHITE));
+		putKielBusStyle("42", new Style(Style.parseColor("#8ba5d6"), Style.WHITE));
+		putKielBusStyle("50", new Style(Style.parseColor("#00a138"), Style.WHITE));
+		putKielBusStyle("51", new Style(Style.parseColor("#00a138"), Style.WHITE));
+		putKielBusStyle("52", new Style(Style.parseColor("#00a138"), Style.WHITE));
+		putKielBusStyle("60S", new Style(Style.parseColor("#92b4af"), Style.WHITE));
+		putKielBusStyle("60", new Style(Style.parseColor("#92b4af"), Style.WHITE));
+		putKielBusStyle("61", new Style(Style.parseColor("#9d1380"), Style.WHITE));
+		putKielBusStyle("62", new Style(Style.parseColor("#9d1380"), Style.WHITE));
+		putKielBusStyle("71", new Style(Style.parseColor("#777e6f"), Style.WHITE));
+		putKielBusStyle("72", new Style(Style.parseColor("#777e6f"), Style.WHITE));
+		putKielBusStyle("81", new Style(Style.parseColor("#00836e"), Style.WHITE));
+		putKielBusStyle("91", new Style(Style.parseColor("#947e62"), Style.WHITE));
+		putKielBusStyle("92", new Style(Style.parseColor("#947e62"), Style.WHITE));
+		putKielBusStyle("100", new Style(Style.parseColor("#d40a11"), Style.WHITE));
+		putKielBusStyle("101", new Style(Style.parseColor("#d40a11"), Style.WHITE));
+		putKielBusStyle("300", new Style(Style.parseColor("#cf94c2"), Style.WHITE));
+		putKielBusStyle("501", new Style(Style.parseColor("#0f3f93"), Style.WHITE));
+		putKielBusStyle("502", new Style(Style.parseColor("#0f3f93"), Style.WHITE));
+		putKielBusStyle("503", new Style(Style.parseColor("#0f3f93"), Style.WHITE));
+		putKielBusStyle("503S", new Style(Style.parseColor("#0f3f93"), Style.WHITE));
+		putKielBusStyle("512", new Style(Style.parseColor("#0f3f93"), Style.WHITE));
+		putKielBusStyle("512S", new Style(Style.parseColor("#0f3f93"), Style.WHITE));
+	}
+
+	private static void putKielBusStyle(String name, Style style) {
+		STYLES.put("AK____|B" + name, style);
```

Comment:
I think the problem with duplicate departures only happens for the lines which are operated by the KVG and Autokraft together (alternatingly), for example lines 501/502. There are also departures which are only listed once with "AK", for example line 900, which is only operated by Autokraft. But the error seems not to be on our side, because they also appear duplicated on the nah.sh website.